File: sources/lua/04_array_sum.lua
-- task 04 array_sum — expected output: 499999500000
-- build: none (interpreted)    run: lua 04_array_sum.lua (PUC Lua 5.4)    also runs on LuaJIT: luajit 04_array_sum.lua

local n = 1000000
local array = {}

for i = 1, n do
    array[i] = i - 1
end

local total = 0
for i = 1, n do
    total = total + array[i]
end

print(total)

File: sources/scala/04_array_sum.scala
// task 04 array_sum — expected output: 499999500000
// build: scalac -release 17 04_array_sum.scala    run: scala Main

object Main {
  def main(args: Array[String]): Unit = {
    val array = new Array[Long](1000000)
    var i = 0
    while (i < 1000000) {
      array(i) = i.toLong
      i += 1
    }
    var total = 0L
    i = 0
    while (i < 1000000) {
      total += array(i)
      i += 1
    }
    println(total)
  }
}
